ckan/logic/action/get.py

```diff
@@ -1875,8 +1877,10 @@ def package_search(context, data_dict):
         else:
             data_source = 'validated_data_dict'
         data_dict.pop('use_default_schema', None)
+
         # return a list of package ids
-        data_dict['fl'] = 'id {0}'.format(data_source)
+        if data_dict.get('fl') not in ('id', 'name'):
+            data_dict['fl'] = 'id {0}'.format(data_source)
```

Comment:
you're expecting the `fl` parameter to be a string value but a list seems like a better choice because solr allows multiple values for `fl`. If you add `fl` to the `default_package_search_schema` you can make sure the value gets converted to a list before it's used here.